lib/customisation.nix

```diff
@@ -309,17 +321,18 @@ rec {
         + "${loc.file}:${toString loc.line}${prettySuggestions (getSuggestions arg)}";
 
       # Only show the error for the first missing argument
-      error = errorForArg (head (attrNames missingArgs));
+      missingError = missingErrorForArg (head (attrNames missingArgs));
 
     in
-    if missingArgs == { } then
-      makeOverridable f allArgs
-    # This needs to be an abort so it can't be caught with `builtins.tryEval`,
-    # which is used by nix-env and ofborg to filter out packages that don't evaluate.
-    # This way we're forced to fix such errors in Nixpkgs,
-    # which is especially relevant with allowAliases = false
+    # These need to be abort so they can't be caught with `builtins.tryEval`,
+    # which is used by nix-env and CI to filter out packages that don't evaluate.
+    # This way we're forced to fix such errors in Nixpkgs.
+    if aliasArgs != { } then
+      abort "lib.customisation.callPackageWith: ${aliasError}"
```

Comment:
If we made aliases become `throw`s when `!config.allowAliases`, would we need this? i.e., they would be filtered out by CI, but would it also just filter out packages that use those aliases? (Just thinking about if we can minimize the number of places that have to know about aliases, for both cleanliness and performance.)